client/engine/engine.go

```diff
@@ -252,22 +252,22 @@ func (e *Engine) handleMessage(message protocols.Message) (EngineEvent, error) {
 
 	}
 
-	for _, entry := range message.LedgerProposals {
+	for _, entry := range message.SortedProposals() {
```

Comment:
```suggestion
        for _, entry := range message.LedgerProposals {
```
I didn't actually want to make this change, here.